From 6b093b5b3c3b6572f5b13c21aa0b8adba32ae6e4 Mon Sep 17 00:00:00 2001
From: Greg McCarthy <greg@gjmccarthy.co.uk>
Date: Mon, 3 Feb 2020 08:50:07 +0000
Subject: [PATCH] Revert "Merge pull request #17287 from lrusak/cpuinfo-next"

This reverts commit e366bab5228259db5981f5475e935eea6c1871b8, reversing
changes made to 4e63657627d4784eb244c1e960aea64a62f644f8.
---
 xbmc/platform/linux/CPUInfoLinux.cpp | 58 +++++++-----------------------------
 xbmc/platform/linux/CPUInfoLinux.h   |  6 ----
 2 files changed, 11 insertions(+), 53 deletions(-)

diff --git a/xbmc/platform/linux/CPUInfoLinux.cpp b/xbmc/platform/linux/CPUInfoLinux.cpp
index 50243a125c..30526d1f8d 100644
--- a/xbmc/platform/linux/CPUInfoLinux.cpp
+++ b/xbmc/platform/linux/CPUInfoLinux.cpp
@@ -81,47 +81,6 @@ CCPUInfoLinux::CCPUInfoLinux()
   if (socPath.Exists())
     m_cpuSoC += " " + socPath.Get<std::string>();
 
-  const std::string freqStr{"/sys/devices/system/cpu/cpu0/cpufreq/scaling_cur_freq"};
-  CSysfsPath freqPath{freqStr};
-  if (freqPath.Exists())
-    m_freqPath = freqStr;
-
-  const std::array<std::string, 3> modules = {
-      "coretemp",
-      "k10temp",
-      "scpi_sensors",
-      "imx_thermal_zone",
-  };
-
-  for (int i = 0; i < 20; i++)
-  {
-    CSysfsPath path{"/sys/class/hwmon/hwmon" + std::to_string(i) + "/name"};
-    if (!path.Exists())
-      continue;
-
-    auto name = path.Get<std::string>();
-
-    if (name.empty())
-      continue;
-
-    for (const auto& module : modules)
-    {
-      if (module == name)
-      {
-        std::string tempStr{"/sys/class/hwmon/hwmon" + std::to_string(i) + "/temp1_input"};
-        CSysfsPath tempPath{tempStr};
-        if (!tempPath.Exists())
-          continue;
-
-        m_tempPath = tempStr;
-        break;
-      }
-    }
-
-    if (!m_tempPath.empty())
-      break;
-  }
-
   m_cpuCount = sysconf(_SC_NPROCESSORS_ONLN);
 
   for (int core = 0; core < m_cpuCount; core++)
@@ -319,20 +279,24 @@ int CCPUInfoLinux::GetUsedPercentage()
 
 float CCPUInfoLinux::GetCPUFrequency()
 {
-  if (m_freqPath.empty())
-    return -1;
+  float value{0};
+  CSysfsPath path("/sys/devices/system/cpu/cpu0/cpufreq/scaling_cur_freq");
+
+  if (path.Exists())
+    value = path.Get<float>() / 1000.0;
 
-  CSysfsPath path{m_freqPath};
-  return path.Get<float>() / 1000.0;
+  return value;
 }
 
 bool CCPUInfoLinux::GetTemperature(CTemperature& temperature)
 {
-  if (m_tempPath.empty())
+  int value{0};
+  CSysfsPath path("/sys/class/hwmon/hwmon0/temp1_input");
+
+  if (!path.Exists())
     return CCPUInfoPosix::GetTemperature(temperature);
 
-  CSysfsPath path{m_tempPath};
-  double value = path.Get<double>() / 1000.0;
+  value = path.Get<int>() / 1000.0;
 
   temperature = CTemperature::CreateFromCelsius(value);
   temperature.SetValid(true);
diff --git a/xbmc/platform/linux/CPUInfoLinux.h b/xbmc/platform/linux/CPUInfoLinux.h
index ed79afc5b6..cc4d95d00d 100644
--- a/xbmc/platform/linux/CPUInfoLinux.h
+++ b/xbmc/platform/linux/CPUInfoLinux.h
@@ -12,8 +12,6 @@
 
 #include "platform/posix/CPUInfoPosix.h"
 
-#include <string>
-
 class CCPUInfoLinux : public CCPUInfoPosix
 {
 public:
@@ -23,8 +21,4 @@ public:
   int GetUsedPercentage() override;
   float GetCPUFrequency() override;
   bool GetTemperature(CTemperature& temperature) override;
-
-private:
-  std::string m_freqPath;
-  std::string m_tempPath;
 };
-- 
2.16.4

